File: backend/app/schemas/excel_export.py
"""Schemas for the Excel financial model export."""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class ExcelExportRequest(BaseModel):
    scenario: str = "base"                       # base | conservative | optimistic | all
    projection_detail: str = "annual"           # monthly | annual | both  (v1 builds annual)
    workbook_type: str = "editable"             # editable | presentation | full
    include_assumptions: bool = True
    include_schedules: bool = True
    include_statements: bool = True
    include_ratios: bool = True
    include_scenarios: bool = True
    include_charts: bool = True
    include_checks: bool = True
    include_text_summary: bool = False
    protect_formulas: bool = True


class ExcelExportFile(BaseModel):
    export_id: str
    project_id: str
    file_name: str
    file_size: int = 0
    scenario: str
    status: str = "ready"                        # ready | failed
    generated_at: datetime
    warnings: list[str] = Field(default_factory=list)
    message: str | None = None


class ExcelExportResponse(ExcelExportFile):
    download_url: str


class ExcelExportPreview(BaseModel):
    project_id: str
    company_name: str
    project_name: str
    currency: str
    period_range: str
    scenario: str
    scenario_label: str
    sheets: list[str] = Field(default_factory=list)
    protect_formulas: bool = True
    data_ready: bool = True
    can_generate: bool = True
    blocking: list[str] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)
    estimated_size_kb: int = 0

File: frontend/src/types/excelExport.ts
/* Types mirroring backend/app/schemas/excel_export.py */
import type { ScenarioKey } from '@/types/incomeStatement'

export type ExcelScenario = ScenarioKey | 'all'
export type ProjectionDetail = 'monthly' | 'annual' | 'both'
export type WorkbookType = 'editable' | 'presentation' | 'full'

export interface ExcelExportRequest {
  scenario: ExcelScenario
  projection_detail: ProjectionDetail
  workbook_type: WorkbookType
  include_assumptions: boolean
  include_schedules: boolean
  include_statements: boolean
  include_ratios: boolean
  include_scenarios: boolean
  include_charts: boolean
  include_checks: boolean
  include_text_summary: boolean
  protect_formulas: boolean
}

export interface ExcelExportResponse {
  export_id: string
  project_id: string
  file_name: string
  file_size: number
  scenario: string
  status: string
  generated_at: string
  warnings: string[]
  message?: string | null
  download_url: string
}

export interface ExcelExportPreview {
  project_id: string
  company_name: string
  project_name: string
  currency: string
  period_range: string
  scenario: string
  scenario_label: string
  sheets: string[]
  protect_formulas: boolean
  data_ready: boolean
  can_generate: boolean
  blocking: string[]
  warnings: string[]
  estimated_size_kb: number
}
